yt_dlp/extractor/projectveritas.py

```diff
@@ -0,0 +1,59 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    try_get,
+    unified_strdate,
+)
+
+
+class ProjectVeritasIE(InfoExtractor):
+    _VALID_URL = r'(?:https?://)(?:www\.)?projectveritas\.com/(?P<type>news|video)/(?P<id>[^/?#]+)'
+    _TESTS = [{
+        'url': 'https://www.projectveritas.com/news/exclusive-inside-the-new-york-and-new-jersey-hospitals-battling-coronavirus/',
+        'info_dict': {
+            'id': '51910aab-365a-5cf1-88f2-8eb1ca5fd3c6',
+            'ext': 'mp4',
+            'title': 'Exclusive: Inside The New York and New Jersey Hospitals Battling Coronavirus',
+            'upload_date': '20200327',
+            'thumbnail': 'md5:6076477fe50b03eb8708be9415e18e1c',
+        }
+    }, {
+        'url': 'https://www.projectveritas.com/video/ilhan-omar-connected-ballot-harvester-in-cash-for-ballots-scheme-car-is-full/',
+        'info_dict': {
+            'id': 'c5aab304-a56b-54b1-9f0b-03b77bc5f2f6',
+            'ext': 'mp4',
+            'title': 'Ilhan Omar connected Ballot Harvester in cash-for-ballots scheme: "Car is full" of absentee ballots',
+            'upload_date': '20200927',
+            'thumbnail': 'md5:194b8edf0e2ba64f25500ff4378369a4',
+        }
+    }]
+
+    def _real_extract(self, url):
+        id, type = self._match_valid_url(url).group('id', 'type')
+        api_url = f'https://www.projectveritas.com/page-data/{type}/{id}/page-data.json'
+        data_json = self._download_json(api_url, id)['result']['data']
+        main_data = data_json.get('video') or data_json.get('post')
+        video_id = main_data['id']
+        thumbnail = try_get(main_data, lambda x: x['image']['ogImage']['src'])
+        mux_asset = main_data.get('muxAsset')
+        if not mux_asset:
+            contents = try_get(main_data, lambda x: x['body']['json']['content']) or []
+            for content in contents:
+                mux_asset = try_get(content, lambda x: x['data']['target']['fields']['muxAsset'])
+                if mux_asset:
+                    break
```

Comment:
```suggestion
        mux_asset = traverse_obj(
            main_data,
            'muxAsset', ('body', 'json', 'content', ..., 'data', 'target', 'fields', 'muxAsset')
            get_all=False, expected_type=dict)
```

`...` means that the key/index at that position can be anything

also consider changing all `try_get(obj, lambda x[a][b])` to `traverse_obj(obj, (a, b))`

Reply:
any particular reason of using traverse_obj over try_get everywhere other than that i won't have to import try_get?

Reply:
no

Reply:
i figured
anyways i have changed try_get to traverse_obj since 1 less import is enough of a reason for me :D